client/auth.py

```diff
@@ -0,0 +1,279 @@
+        """
+        Gets the user info by the user name.
+        """
+        req = AuthUserGetRequest(name=name)
+        request_with_token = RequestWithToken(token=self.token, auth_user_get_request=req)
+        er = await self.handle_req(req=request_with_token, use_fast_path=True)
+        return er.auth_user_get_response
+
+    async def user_list(self) -> AuthUserListResponse:
+        """
+        Lists all users.
+        """
+        request_with_token = RequestWithToken(token=self.token, auth_user_list_request=AuthUserListRequest())
+        er = await self.handle_req(req=request_with_token, use_fast_path=True)
+        return er.auth_user_list_response
+
+    async def user_delete(self, name: str) -> AuthUserDeleteResponse:
+        """
+        Deletes the given key from the key-value store.
+        """
+        req = AuthUserDeleteRequest(name=name)
+        request_with_token = RequestWithToken(token=self.token, auth_user_delete_request=req)
+        er = await self.handle_req(req=request_with_token, use_fast_path=False)
+        return er.auth_user_delete_response
+
+    async def user_change_password(self, name: str, password: str) -> AuthUserChangePasswordResponse:
+        """
+        Change password for an user.
+        """
+        req = AuthUserChangePasswordRequest(name=name, password=password)
+        if req.password == "":
+            msg = "role name is empty"
+            raise Exception(msg)
+        hashed_password = self.hash_password(req.password)
+        req.hashedPassword = hashed_password
+        req.password = ""
+
+        request_with_token = RequestWithToken(token=self.token, auth_user_change_password_request=req)
+        er = await self.handle_req(req=request_with_token, use_fast_path=False)
+        return er.auth_user_change_password_response
+
+    async def user_grant_role(self, user: str, role: str) -> AuthUserGrantRoleResponse:
+        """
+        Grant role for an user.
+        """
+        req = AuthUserGrantRoleRequest(user=user, role=role)
+        request_with_token = RequestWithToken(token=self.token, auth_user_grant_role_request=req)
+        er = await self.handle_req(req=request_with_token, use_fast_path=False)
+        return er.auth_user_grant_role_response
+
+    async def user_revoke_role(self, name: str, role: str) -> AuthUserRevokeRoleResponse:
+        """
+        Revoke role for an user.
+        """
+        req = AuthUserRevokeRoleRequest(name=name, role=role)
+        request_with_token = RequestWithToken(token=self.token, auth_user_revoke_role_request=req)
+        er = await self.handle_req(req=request_with_token, use_fast_path=False)
+        return er.auth_user_revoke_role_response
+
+    async def role_add(self, name: str) -> AuthRoleAddResponse:
+        """
+        Adds role.
+        """
+        req = AuthRoleAddRequest(name=name)
+        if req.name == "":
+            msg = "role name is empty"
+            raise Exception(msg)
```

Comment:
Please check before constructing the request.

Reply:
Please double-check the problem in the whole code base.

Reply:
I've modified a similar problem that exists whole code.